lib/Dialect/Torch/Transforms/RecomposeComplexOps.cpp

```diff
@@ -71,6 +72,56 @@ class RecomposeSliceCopy_ : public OpRewritePattern<AtenCopy_Op> {
     return success();
   }
 };
+
+class RecomposeSelectFill_ : public OpRewritePattern<AtenFill_TensorOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenFill_TensorOp op,
+                                PatternRewriter &rewriter) const override {
+    if (!op.getSelf().getDefiningOp() ||
+        !isa<AtenSelectIntOp>(op.getSelf().getDefiningOp()))
+      return failure();
+    auto selectOp = cast<AtenSelectIntOp>(op.getSelf().getDefiningOp());
+
+    // Get indices
+    int64_t dim;
+    if (!matchPattern(selectOp.getDim(), m_TorchConstantInt(&dim)))
+      return failure();
+
+    Value noneVal = rewriter.create<ConstantNoneOp>(op.getLoc());
+    Value falseVal = rewriter.create<ConstantBoolOp>(op.getLoc(), false);
+
+    // Create IndexPut_Op
+    // Convert indexNum to indexTensor for the selectOp
+    BaseTensorType selectOutTy = selectOp.getType().template cast<BaseTensorType>();
+    SmallVector<int64_t> empty;
+    auto dtype =
+        getTypeForTorchType(selectOp.getContext(), selectOp.getIndex().getType());
+    Type emptyTensorType = selectOutTy.getWithSizesAndDtype(llvm::ArrayRef(empty), dtype);
+    Value indexTensor = rewriter.create<PrimNumToTensorScalarOp>(
+        selectOp.getLoc(), emptyTensorType, selectOp.getIndex());
+
+    // Create indicesVector for IndexPut_Op by TorchNone and indexTensor
+    BaseTensorType tensorType = op->getResultTypes()[0].cast<BaseTensorType>();
+    SmallVector<Value> indicesVector;
+    for (auto i = 0; i < dim-1; i++) {
```

Comment:
This for loop can be replaced with `SmallVector<Value> indicesVector(dim-1, noneVal);`